ocaml/testsuite/tests/typing-modes/portable-contend.ml

```diff
@@ -225,15 +301,33 @@ Error: This function when partially applied returns a value which is nonportable
        but expected to be portable.
 |}]
 
+(* closing over shared gives nonportable *)
+let foo : 'a @ shared portable -> (unit -> unit) @ portable = fun a () -> ()
+[%%expect{|
+Line 1, characters 62-76:
+1 | let foo : 'a @ shared portable -> (unit -> unit) @ portable = fun a () -> ()
```

Comment:
This surprises me. The returned closure does not use the fact that `a` is shared. So (I think) we can submode the `a` to be `contended` and then close over the contended `a` without make the closure `nonportable`.

Reply:
Indeed, I see what you mean. This test has the same behavior as: 
`let foo : 'a @ uncontended portable -> (unit -> unit) @ portable = fun a () -> ()`
which also gives an error, even though `a` (here `uncontended`) is not used.

Reply:
OK. We don't have to fix this here. But I think this is another good time to add `(* CR modes: This should be allowed *)` to encourage us to return.